### PR TITLE
[PLAT-715] Keycloak client framework

### PR DESCRIPTION
**Pull request by carrala**
- Adds keycloak client.
- Implements GetToken and supporting methods.
- Tests in keycloakClient coming soon!

### CODE REVIEW

**Review by galxy25 — APPROVED** [review]

**Review by rosasck — APPROVED** [review]
👏

**Review by efabens — APPROVED** [review]

File: keycloakClient/keycloakClient_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	keycloakClientConfig := Config{
+		AddrTokenProvider: cyclopsServiceHost,
+		AddrAPI:           cyclopsServiceHost,
+		Timeout:           HTTPTimeout,
+	}
+	kcClient, err := New(keycloakClientConfig)
+	if err != nil {
+		t.Fatalf("Failure creating keycloak client with err: %+v", err)
+	}
+	_, err = kcClient.GetToken(keycloakMasterRealm, keycloakUsername, keycloakPassword)
```
Comment (galxy25) [inline keycloakClient/keycloakClient_test.go:29]: nit: wouldn't hurt to also test/assert that the returned token is a non empty string
